pkg/ring/kv/etcd/etcd.go

```diff
@@ -34,10 +41,49 @@ func (cfg *Config) RegisterFlagsWithPrefix(f *flag.FlagSet, prefix string) {
 	f.Var((*flagext.StringSlice)(&cfg.Endpoints), prefix+"etcd.endpoints", "The etcd endpoints to connect to.")
 	f.DurationVar(&cfg.DialTimeout, prefix+"etcd.dial-timeout", 10*time.Second, "The dial timeout for the etcd connection.")
 	f.IntVar(&cfg.MaxRetries, prefix+"etcd.max-retries", 10, "The maximum number of retries to do for failed ops.")
+	// Get certs from flag params and set config TLS, if present
+	f.StringVar(&cfg.CertFile, prefix+"etcd.cert-file", "", "The cert file dir location")
+	f.StringVar(&cfg.KeyFile, prefix+"etcd.key-file", "", "The key file dir location")
+	f.StringVar(&cfg.TrustedCAFile, prefix+"etcd.client-ca-file", "", "The trusted ca file directory location")
+}
+
+// SetTLS receives cert file locations and builds transport TLS info
+func (cfg *Config) SetTLS() error {
```

Comment:
I would merge `SetTLS()` logic into `GetTLS()`. No need to store TLS config in `cfg.TLS`. You can just have `GetTLS()` generating it on the fly and then returning `*transport.TLSInfo`.